packages/atomic/cypress/fixtures/test-fixture.ts

```diff
@@ -0,0 +1,129 @@
+type SearchInterface = HTMLElement & {
+  initialize: (opts: {
+    accessToken: string;
+    organizationId: string;
+  }) => Promise<void>;
+  executeFirstSearch: () => void;
+};
+
+export type TestFeature = (e: TestFixture) => void | Promise<void>;
+
+export type TagProps = Record<string, string>;
+
+export class TestFixture {
+  private urlParams: {key: string; value: string}[] = [];
+  private aliases: TestFeature[] = [];
+  private testURL = 'http://localhost:3333/pages/test.html';
+  private execFirstSearch = true;
+
+  constructor() {
+    cy.visit(this.testURL).injectAxe();
+    this.intercept();
+
+    cy.document().then((doc) => {
+      const searchInterface = doc.createElement(
+        'atomic-search-interface'
+      ) as SearchInterface;
+      doc.body.appendChild(searchInterface);
+      cy.get('atomic-search-interface').as(this.elementAliases.SearchInterface);
+    });
+  }
+
+  public with(feat: TestFeature) {
+    feat(this);
+    return this;
+  }
+
+  public withURLParam(key: string, value: string) {
```

Comment:
I feel there are two kinds of categories of things handled by this class:
1. Html configuration, which is done using `with` and injecting DOM nodes.
2. Run-time configuration, in this case, the url of the browser.

Would it make sense to have this class just focused on the markup?
And then I imagine we can use cypress for browser-based things like setting the url, rather than creating a wrapper around it?

Reply:
Hum yeah, I guess I could remove the things about URL parameter indeed/look for things made available with cypress.